src/codeu/chat/client/commandline/Chat.java

```diff
@@ -46,11 +46,19 @@ public Chat(Context context) {
   // the system wants to exit, the function will return false.
   //
   public boolean handleCommand(String line) {
-
+/*
     final Scanner tokens = new Scanner(line.trim());
-
     final String command = tokens.hasNext() ? tokens.next() : "";
-
+*/
+  final List<String> args = new ArrayList<>();
+  final Tokenizer tokenizer = new Tokenizer(line);
+  for(String token = tokenizer.next(); token != null; token = tokenizer.next())
```

Comment:
This is the part of code reviews where I introduce you to style nit-picking. ;-)

The code review guidelines point to the Google Java Style Guide, which has opinions about whitespace and what line your brackets go on and so forth. In particular, keywords like `for` are supposed to be followed by a space (https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace) and opening curly-braces are supposed to appear on the same line as the `if` or `for` that they belong to (https://google.github.io/styleguide/javaguide.html#s4.1.2-blocks-k-r-style).